### PR TITLE
return to main mktplace page from rez page

### PR DESCRIPTION
**Pull request by MandatoryMania**
Once an item is purchased, allows "Continue Shopping" to return to main Marketplace page. 
https://highfidelity.fogbugz.com/f/cases/10229/Confirmation-page-after-a-purchase

Test Plan

1.  Open Marketplace 
2.  Buy an item or obtain a free item
3.  Confirm purchase
4.  On the Thank You page (with the REZ IT or WEAR IT button): 
    Click on "CONTINUE SHOPPING"
5. You should arrive at the main page of Marketplace

### CODE REVIEW

File: scripts/system/marketplaces/marketplaces.js [hunk 1]
```diff
@@ -421,7 +421,8 @@
                 break;
             case 'checkout_itemLinkClicked':
             case 'checkout_continueShopping':
-                tablet.gotoWebScreen(MARKETPLACE_URL + '/items/' + message.itemId, MARKETPLACES_INJECT_SCRIPT_URL);
+                //tablet.gotoWebScreen(MARKETPLACE_URL + '/items/' + message.itemId, MARKETPLACES_INJECT_SCRIPT_URL);
```
Comment (davidkelly) [inline scripts/system/marketplaces/marketplaces.js]: just remove this, no need to comment it out.

File: scripts/system/marketplaces/marketplaces.js [hunk 2]
```diff
@@ -421,7 +421,8 @@
                 break;
             case 'checkout_itemLinkClicked':
```
Comment (zfox23) [inline scripts/system/marketplaces/marketplaces.js]: This case needs to be correctly handled now.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5455/HighFidelity-Beta-PR11943-ed76ca8018df6eeee3e96482f594cdda3e357cdb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5455/HighFidelity-Beta-PR11943-ed76ca8018df6eeee3e96482f594cdda3e357cdb.dmg

File: scripts/system/marketplaces/marketplaces.js (hunk 1 above)
Comment (MandatoryMania) [reply scripts/system/marketplaces/marketplaces.js]: Done

File: scripts/system/marketplaces/marketplaces.js (hunk 2 above)
Comment (MandatoryMania) [reply scripts/system/marketplaces/marketplaces.js]: Done, and thank you for reminding me of that!

Comment (davidkelly) [reply scripts/system/marketplaces/marketplaces.js]: oh yes -- this still should go to the item itself

Comment (MandatoryMania) [reply scripts/system/marketplaces/marketplaces.js]: Yes, that's where it goes now.  Earlier it was a fall-through (dangerous).

**Comment by zfox23** [comment]
I'm not seeing the changes, @MandatoryMania - did you push?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5475/HighFidelity-Beta-PR11943-72ed3b077df4328d4b28fbbe7c3956787acb8c57.dmg

**Comment by YuppyPlays** [comment]
Awaiting windows build for QA.

**Comment by marko8904** [comment]
@MandatoryMania you'll need to add your email from one of the commits to you Github account. commit 72ed3b0 is the one containing the email that's not associated. This prevents the CLA from being marked as signed.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5485/HighFidelity-Beta-PR11943-72ed3b077df4328d4b28fbbe7c3956787acb8c57.exe

**Comment by YuppyPlays** [comment]
Testing.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5485/HighFidelity-Beta-PR11943-72ed3b077df4328d4b28fbbe7c3956787acb8c57.dmg

**Comment by YuppyPlays** [comment]
Testing complete:

Open Marketplace **Check**
Buy an item or obtain a free item **Need a specific item to test as I've gone through quite a few on the store and everytime I find one that is 0 HFC & an entity so it shows in Purchases history, I get the following kind of error**
image

Confirm purchase **Fail, Please see image above**
On the Thank You page (with the REZ IT or WEAR IT button): **Fail, Please see image above**
Click on "CONTINUE SHOPPING" **Fail, Please see image above**
You should arrive at the main page of Marketplace **Fail, Please see image above**

**Fail**

I can carry on with this if I can get an item to test that will definitely work as every one I've tried that seems to fit the bill doesn't seem to work.

**Comment by MandatoryMania** [comment]
I think I know what happened.  My acct on GitHub has hisham@highfidelity.io <mailto:hisham@highfidelity.io> as one of my emails.  The commit is asking that hisham@highfidelity.com <mailto:hisham@highfidelity.com> be added.  I was told they point to the same place.  Anyway, I’ll add the latter to my GitHub acct. 

thx, 

Hisham

**Comment by MandatoryMania** [comment]
@YuppyPlays 
Regarding your comments on PR11948, that error you are seeing is due to: attempting to buy from an item whose author has not set up their wallet.  We should not have any of those.  So can you indicate what items you got this result for?  BTW, try purchasing Green Chair, Samurai Helmet, Basic Torch, they wall work fine.

**Comment by YuppyPlays** [comment]
@MandatoryMania 
As discussed I shall list out the items I found and try one of those purchases within test to confirm QA.

**Comment by YuppyPlays** [comment]
Testing.

[Will also list out the store items with that error]

**Comment by YuppyPlays** [comment]
Testing complete:

Open Marketplace **Check**
Buy an item or obtain a free item **Check, in the end I used Samurai Helmet on a wallet file I had with funds**
Confirm purchase **Check**
On the Thank You page (with the REZ IT or WEAR IT button): **Check**
Click on "CONTINUE SHOPPING" **Check**
You should arrive at the main page of Marketplace **Check**

**Pass**

**regarding the items, I spoke to Chang and it's been boiled down to my account somehow allowing me to setup a new wallet under it even though I have done this previously within other builds (which I honestly did not realise was not meant to happen, explains a lot!), bug filed: https://highfidelity.fogbugz.com/f/cases/10332/New-wallet-working-multiple-times-under-same-account**

Thank you @changxc for the guidance on that :)

**Pull request merged**